packages/office-addin-dev-settings/src/sideload.ts

```diff
@@ -92,6 +92,40 @@ async function generateSideloadFile(app: OfficeApp, manifest: ManifestInfo): Pro
   return pathToWrite;
 }
 
+/**
+ * Create an Office document url with query params which can be opened
+ * to register an Office add-in in Office Online.
+ * @param documentUrl Office Online document url
+ * @param manifestPath Path to the manifest file for the Office Add-in.
+ * @param port Port number for the dev-server.
+ * @param isTest Indicates whether to append test query param to suppress Office Online dialogs.
+ * @returns Document url with query params appended.
+ */
+export async function generateSideloadUrl(documentUrl: string, manifestPath: string, port: number, isTest:boolean = false): Promise<string> {
+  const regEx: RegExp = /^.*[\\\/]/;
+  const devServerPortQueryParam = "&wdaddindevserverport=";
+  const manifestFileQueryParam = "&wdaddinmanifestfile=";
+  const manifestGuidQueryParam = "&wdaddinmanifestguid=";
+  const testQueryParam = "&wdaddintest=true";
+  const manifest = await readManifestFile(manifestPath);
+  const appsInManifest: OfficeApp[] = getOfficeAppsForManifestHosts(manifest.hosts);
+  appsInManifest.forEach(element => {
+    if (element === OfficeApp.Outlook || element === OfficeApp.Project) {
```

Comment:
Shouldn't this be done before calling this function? It would be clearer to have this check be separate and not doing when adding the query parameters for sideloading to the web.

Reply:
sure, we can move this up in the stack